### PR TITLE
Fix compiler warnings

### PR DESCRIPTION
**Pull request by markusdregi**
**Task**
_Compiling libecl yields numerous warnings._

**Approach**
_Make the compiler happy by fixing warnings._

**Pre un-WIP checklist**
- [x] Statoil tests pass locally

### CODE REVIEW

File: lib/CMakeLists.txt [hunk 1]
```diff
@@ -50,8 +50,17 @@ if (MSVC)
     configure_file(include/ert/util/msvc_stdbool.h ert/util/stdbool.h)
 endif ()
 
-foreach (type int double bool long time_t size_t float)
```
Comment (joakim-hove) [inline lib/CMakeLists.txt]: Hmmm - we certainly need the `bool`  variety; at least in Python.

File: lib/util/vector_template.c [hunk 2]
```diff
@@ -871,14 +871,24 @@ void @TYPE@_vector_set_all(@TYPE@_vector_type * vector , @TYPE@ value) {
 void @TYPE@_vector_init_range(@TYPE@_vector_type * vector , @TYPE@ value1 , @TYPE@ value2 , @TYPE@ delta) {
   @TYPE@_vector_reset( vector );
   if (delta != 0) {
-    if (((delta > 0) && (value2 > value1)) || ((delta < 0) && (value2 < value1))) {
+    bool valid_range = delta > 0 && value2 > value1;
+    #if @UNSIGNED_TYPE@ == 0
```
Comment (joakim-hove) [inline lib/util/vector_template.c]: Quelching warnings this way was elegant!

Comment (joakim-hove) [inline lib/util/vector_template.c]: The preprocessor languare has boolean-logic; i.e. this test could be rewritten
```
#if @SIGNED_TYPE@
    valid_range |= ...
```

Comment (markusdregi) [inline lib/util/vector_template.c]: Good point! I'll fix it

**Pull request merged**